src/Features/Core/Portable/MetadataAsSource/DecompilationMetadataAsSourceFileProvider.cs

```diff
@@ -184,6 +184,10 @@ public DecompilationMetadataAsSourceFileProvider(IImplementationAssemblyLookupSe
                     GlobalAssemblyCache.Instance.ResolvePartialName(fullAssemblyName, out assemblyLocation, preferredCulture: CultureInfo.CurrentCulture);
                     isReferenceAssembly = assemblyLocation is null;
                 }
+                catch (IOException)
+                {
+                    // If we get an IO exception we can safely ignore it, and the system will show the metadata view of the reference assembly.
```

Comment:
@davidwengier Which type of IOException were we getting here?

Reply:
FileLoadException I believe